File: src/game/events/EnemyDespawnedGameEvent.java
package game.events;

import game.Bomberman;

public class EnemyDespawnedGameEvent implements GameEvent {
    @Override
    public void invoke(Object arg) {
        if (!Bomberman.getMatch().getGameState()) return;

        System.out.println("KILLED ENEMY! " + Bomberman.getMatch().getEnemiesAlive());
        if(Bomberman.getMatch().getEnemiesAlive() == 0) {
            new AllEnemiesEliminatedGameEvent().invoke(null);
        }
    }
}

File: src/game/engine/events/game/EnemyDespawnedGameEvent.kt
package game.engine.events.game

import game.Bomberman
import game.engine.events.models.GameEvent

class EnemyDespawnedGameEvent : GameEvent {
    override fun invoke(arg: Any?) {
        if (!Bomberman.getMatch().gameState)
            return
        Bomberman.getMatch().currentLevel!!.eventHandler.onEnemyDespawned()
    }
}